clinica/iotools/converters/oasis3_to_bids/oasis3_to_bids.py

```diff
@@ -0,0 +1,165 @@
+        bids_ids = bids.get_bids_subjs_list(bids_dir)
+
+        iotools_folder = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
+        clinic_specs_path = path.join(iotools_folder, "data", "clinical_specifications")
+
+        # -- Creation of modality agnostic files --
+        bids.write_modality_agnostic_files("OASIS-3", bids_dir)
+
+        # --Create participants.tsv--
+        participants_df = bids.create_participants_df(
+            "OASIS3", clinic_specs_path, clinical_data_dir, bids_ids
+        )
+
+        # Replace the values of the diagnosis_bl column
+        participants_df["diagnosis_bl"].replace([0.0, np.nan], "CN", inplace=True)
+        participants_df["diagnosis_bl"].replace(
+            [0.5, 1.0, 1.5, 2.0], "AD", inplace=True
+        )
+        # Following line has no sense
+        # participants_df['diagnosis_bl'].replace(participants_df['diagnosis_bl']>0.0, 'AD', inplace=True)
+        participants_df = participants_df.fillna("n/a")
+        participants_df.to_csv(
+            path.join(bids_dir, "participants.tsv"),
+            sep="\t",
+            index=False,
+            encoding="utf-8",
+        )
+
+        # --Create sessions files--
+        sessions_dict = bids.create_sessions_dict(
+            clinical_data_dir, "OASIS3", clinic_specs_path, bids_ids, "ID"
+        )
+        for y in bids_ids:
+            for z in sessions_dict[y].keys():
+                if sessions_dict[y][z]["diagnosis"] > 0:
+                    sessions_dict[y][z]["diagnosis"] = "AD"
+                else:
+                    sessions_dict[y][z]["diagnosis"] = "CN"
+
+        bids.write_sessions_tsv(bids_dir, sessions_dict)
+
+        # --Create scans files--
+        # Note: We have no scans information for OASIS
+        scans_dict = bids.create_scans_dict(
+            clinical_data_dir,
+            "OASIS",
+            clinic_specs_path,
+            bids_ids,
+            "ID",
+            "",
+            sessions_dict
+        )
+        bids.write_scans_tsv(bids_dir, bids_ids, scans_dict)
+
+    def convert_images(self, source_dir, dest_dir):
+        """Convert T1w images to BIDS.
+
+        Args:
+            source_dir: path to the OASIS dataset
+            dest_dir: path to the BIDS directory
+        """
+        from os import path
+        from glob import glob
+        import os
+
+        def convert_single_subject(subj_folder):
+            import os
+            import re
+            import shutil
+
+            def copy_file(source, target):
+                try:
+                    new_filename = re.sub("OAS3", "OASIS3", path.basename(source))
+                    new_filename = re.sub("_echo-", "_run-", new_filename)
+                    if re.search(".*_pet.(nii.gz|json)", new_filename) is not None:
+                        new_filename = re.sub("task-rest_", "", new_filename)
+                        new_filename = re.sub("acq", "trc", new_filename)
```

Comment:
You don't need `re.sub` for all this. `str.replace` is enough.

Reply:
Also, I am not quite sure what you are doing here. Pehaps a comment would help.

Reply:
(After discussions with @14thibea , it was requested to )
"# according to the new BIDS convention for PET, to use the keyword trc instead of acq. 
# in the same vein to remove the optional task keyword which is useless because it is always rest
# that the minIP files look like the list given on page 10 of this document https://www.oasis-brains.org/files/OASIS-3_Imaging_Data_Dictionary_v1.8.pdf --> so we would put the swi and minIP files in a swi/ folder, and we would make the bids-validator ignore it since it is not in the standard."

Maybe a proofreading from her as well ... to see if I did something wrong ;)

Reply:
Thanks for clarifying. Then please add a comment as a preamble to this code explaining all this to make the necessary context explicit.